dll/win32/uxtheme/nonclient.c

```diff
@@ -210,12 +211,11 @@ ThemeEndBufferedPaint(PDRAW_CONTEXT pcontext, int x, int y, int cx, int cy)
     pcontext->hDC = pcontext->hDCScreen;
 }
 
-void ThemeCalculateCaptionButtonsPos(HWND hWnd, HTHEME htheme)
+static void ThemeCalculateCaptionButtonsPosEx(WINDOWINFO wi, HWND hWnd, HTHEME htheme, INT buttonHeight)
```

Comment:
For performance reasons would it make sense to pass `WINDOWINFO` as pointer?

Reply:
I think it would be,
but if that is worth it depends on how often this function is called (and I don't know this).

Reply:
To clarify...are you saying that passing `WINDOWINFO` might be *slower* depending on how often the function is called? Or just that passing `WINDOWINFO` might result in a neglibible performance gain (if any)?

Reply:
What Mark is implicitly telling you is that passing a large structure by pointer instead of as a full copy on the stack should be better.

Reply:
Ohhhh, okay...yeah, that makes sense now. Done.